src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -1451,7 +1451,7 @@ static grpc_error* begin_parse_string(grpc_chttp2_hpack_parser* p,
                                       const uint8_t* cur, const uint8_t* end,
                                       uint8_t binary,
                                       grpc_chttp2_hpack_parser_string* str) {
-  if (!p->huff && binary == NOT_BINARY &&
+  if ((end > cur) && !p->huff && binary == NOT_BINARY &&
```

Comment:
I think there's a more fundamental question here that needs to be addressed: why are we trying to parse a string when cur == end? In other words, there's nothing to parse...

Let me take a look at the stack traces and the code more to figure out why we even attempt this...

Reply:
Agreed, I would rather that we use this as an opportunity to clean up the code and define its invariants rather than rush a review

Reply:
@vjpai @equinox1993 @yang-g I reviewed the stack traces and the code, and I still do not understand how we get into a state where we are trying to parse an empty string. However, what do you think about adding an end condition to the beginning of this function? 
`  if (cur == end) {
    p->state = begin_parse_string; // I'm not sure if this is the correct state to set here
    return GRPC_ERROR_NONE;
  } `
This might fix the fuzzer issue and would prevent us from going down the code path below the if branch.

I do think someone should look at the code and figure out its invariants, but it's so unnecessarily complex that I don't think anyone has time to / wants to.

Reply:
Yeah. This is what I first tried locally and it fixes the error too.

Reply:
@vjpai, would you mind changing the code to the suggestion? I think it makes more sense to add in an end condition here (like elsewhere in the code) rather than just the "end > cur" check